Sources/BraintreeCore/BTAPIClient.m

```diff
@@ -273,75 +274,55 @@ - (void)fetchPaymentMethodNonces:(BOOL)defaultFirst completion:(void (^)(NSArray
-    //       is processed independently. Thus, the check for cached configuration and the fetch is an
-    //       atomic operation with respect to other calls to this method.
-    //
-    // Note: Uses dispatch_semaphore to block the configuration queue when the configuration fetch
-    //       request is waiting to return. In this context, it is OK to block, as the configuration
-    //       queue is a background queue to guarantee atomic access to the remote configuration resource.
-    dispatch_async(self.configurationQueue, ^{
-        __block NSError *fetchError;
-
-        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
-        __block BTConfiguration *configuration;
-        NSString *configPath = @"v1/configuration"; // Default for tokenizationKey
-        if (self.clientToken) {
-            configPath = [self.clientToken.configURL absoluteString];
-        }
-        [self.configurationHTTP GET:configPath parameters:@{ @"configVersion": @"3" } shouldCache:YES completion:^(BTJSON * _Nullable body, NSHTTPURLResponse * _Nullable response, NSError * _Nullable error) {
-            if (error) {
-                fetchError = error;
-            } else if (response.statusCode != 200) {
-                NSError *configurationDomainError =
-                [NSError errorWithDomain:BTAPIClientErrorDomain
-                                    code:BTAPIClientErrorTypeConfigurationUnavailable
-                                userInfo:@{
-                                           NSLocalizedFailureReasonErrorKey: @"Unable to fetch remote configuration from Braintree API at this time."
-                                           }];
-                fetchError = configurationDomainError;
-            } else {
-                configuration = [[BTConfiguration alloc] initWithJSON:body];
-                if (!self.braintreeAPI) {
-                    NSURL *apiURL = [configuration.json[@"braintreeApi"][@"url"] asURL];
-                    NSString *accessToken = [configuration.json[@"braintreeApi"][@"accessToken"] asString];
-                    self.braintreeAPI = [[BTAPIHTTP alloc] initWithBaseURL:apiURL accessToken:accessToken];
-                }
-                if (!self.http) {
-                    NSURL *baseURL = [configuration.json[@"clientApiUrl"] asURL];
-                    if (self.clientToken) {
-                        self.http = [[BTHTTP alloc] initWithBaseURL:baseURL authorizationFingerprint:self.clientToken.authorizationFingerprint];
-                    } else if (self.tokenizationKey) {
-                        self.http = [[BTHTTP alloc] initWithBaseURL:baseURL tokenizationKey:self.tokenizationKey];
-                    }
+    //   - If cachedConfiguration is not present, fetch it and cache the successful response
+    //   - If fetching fails, return error
+    __block NSError *fetchError;
+    __block BTConfiguration *configuration;
```

Comment:
Would it be possible to move these into the closure to avoid `__block` scope?

Reply:
✅